src/main/java/hudson/plugins/brakeman/BrakemanPublisher.java

```diff
@@ -101,7 +101,17 @@ public BuildResult publishReport(final Run<?, ?> build, final FilePath workspace
 			String output = brakemanOutput.readToString();
 
 			ParserResult project = new ParserResult(workspace);
-			this.scan(output, project);
+
+			AbstractBrakemanScanner scanner;
+
+			if(this.outputFile.endsWith(".json")) {
+				scanner = new BrakemanJSONScanner();
```

Comment:
I don't think I know exactly what you mean. This basically triggers the different readers. The JSON reader will read both warnings and ignored warnings because the output file includes that. The tabs scanner will just have warnings because it only includes warnings.